src/main/java/io/xpring/xrpl/model/XrpCheckCreate.java

```diff
@@ -0,0 +1,116 @@
+   *         since the Ripple Epoch.
+   */
+  Optional<Integer> expiration();
+
+  /**
+   * (Optional) Arbitrary 256-bit hash representing a specific reason or identifier for this Check.
+   *
+   * @return A {@link String} containing an arbitrary 256-bit hash representing a specific reason or identifier for
+   *         this Check.
+   */
+  Optional<String> invoiceID();
+
+  /**
+   * Maximum amount of source currency the Check is allowed to debit the sender, including transfer fees on non-XRP
+   * currencies.
+   * <p>
+   * The Check can only credit the destination with the same currency (from the same issuer, for non-XRP currencies).
+   * For non-XRP amounts, the nested field names MUST be lower-case.
+   * </p>
+   * @return A {@link XrpCurrencyAmount} representing the maximum amount of source currency the Check is allowed
+   *         to debit the sender, including transfer fees on non-XRP currencies.
+   */
+  Optional<XrpCurrencyAmount> sendMax();
+
+  /**
+   * Constructs an {@link XrpCheckCreate} from a CheckCreate protocol buffer.
+   *
+   * @param checkCreate A {@link CheckCreate} (protobuf object) whose field values will be used to construct an XrpCheckCreate
+   * @return An {@link XrpCheckCreate} with its fields set via the analogous protobuf fields.
+   * @see "https://github.com/ripple/rippled/blob/3d86b49dae8173344b39deb75e53170a9b6c5284/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L145"
+   */
+  static XrpCheckCreate from(CheckCreate checkCreate, XrplNetwork xrplNetwork) {
+    // Destination is required
+    if (!checkCreate.hasDestination() || checkCreate.getDestination().getValue().getAddress().isEmpty()) {
+      return null;
+    }
+    final String destination = checkCreate.getDestination().getValue().getAddress();
+
+    Optional<Integer> destinationTag = checkCreate.hasDestinationTag()
+        ? Optional.of(checkCreate.getDestinationTag().getValue())
+        : Optional.empty();
+
+    ClassicAddress classicAddress = ImmutableClassicAddress.builder()
+        .address(destination)
+        .tag(destinationTag)
+        .isTest(xrplNetwork == XrplNetwork.TEST || xrplNetwork == XrplNetwork.DEV)
+        .build();
+
+    final String destinationXAddress = Utils.encodeXAddress(classicAddress);
+
+    Optional<Integer> expiration = checkCreate.hasExpiration()
+        ? Optional.of(checkCreate.getExpiration().getValue())
+        : Optional.empty();
+
+    Optional<String> invoiceID = checkCreate.hasInvoiceId()
+        ? Optional.of(checkCreate.getInvoiceId().getValue().toString())
+        : Optional.empty();
+
+    // If the sendMax field is set, it must be able to be transformed into an XrpCurrencyAmount.
+    Optional<XrpCurrencyAmount> sendMax = Optional.empty();
+    if (checkCreate.hasSendMax()) {
+      sendMax = Optional.ofNullable(XrpCurrencyAmount.from(checkCreate.getSendMax().getValue()));
+      if (!sendMax.isPresent()) {
+        return null;
+      }
+    }
```

Comment:
i took this from another place that had a `sendMax`

Reply:
seems legit